lib/install/management/update.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package management
+
+import (
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/vic/pkg/trace"
+)
+
+type State int
+
+// ruleset ID from /etc/vmware/firewall/service.xml
+const RulesetID string = "vSPC"
+const (
+	enable  State = iota
+	disable       = iota
+)
+
+// EnableFirewall enables the ruleset on the target, allowing VIC backchannel traffic
+func (d *Dispatcher) EnableFirewallRuleset() error {
+	defer trace.End(trace.Begin(""))
+	return d.modifyFirewall(enable)
+}
+
+// DisableFirewall disables the ruleset on the target, denying VIC backchannel traffic
+func (d *Dispatcher) DisableFirewallRuleset() error {
+	defer trace.End(trace.Begin(""))
+	return d.modifyFirewall(disable)
+}
+
+// modifyFirewall sets the state of the firewall ruleset specified by RulesetID
+func (d *Dispatcher) modifyFirewall(state State) error {
+	defer trace.End(trace.Begin(""))
+	var err error
+	var hosts []*object.HostSystem
+	var s string
+
+	if state == enable {
+		s = "enable"
+	} else {
+		s = "disable"
+	}
+
+	log.Debugf("cluster: %s", d.session.Cluster)
+
+	hosts, err = d.session.Cluster.Hosts(d.ctx)
+	if err != nil {
+		return err
+	}
+
+	if len(hosts) == 0 {
+		log.Infof("No hosts to modify")
+		return nil
+	}
+
+	for _, host := range hosts {
+		fs, err := host.ConfigManager().FirewallSystem(d.ctx)
+		if err != nil {
+			log.Errorf("Failed to get firewall system for host %q: %s", host.Name(), err)
+			return err
+		}
+
+		if state == enable {
```

Comment:
cause you need to switch the enable/disable anyhow in the loop, you might not need the test at line52, you can just log out in side of this test.